src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs

```diff
@@ -205,7 +205,12 @@ private static (CSharpType PropertyType, CSharpType ValueType) GetPropertyDefaul
             }
 
             var valueType = propertyType;
-            if (propertyType.IsValueType && !property.IsRequired)
+            if (propertyType.IsValueType &&
+                !property.IsRequired &&
+                !isPropertyBag) // For property bag properties, we keep the original type nullability, because in method parameters, we have two types of optional parameters
+                // "int a = 1, in? b = 3", if we will transform all optional parameters as model nullable properties, then there will be compilation error in the property bag
+                // expansion logic, e.g. "Foo(options.A)" won't match "Foo(int a = 1)", since "options.A" is of type int?
+                // Another way to resolve this issue is to change the expansion expression like "Foo(options.A ?? 1)", which is urgly.
```

Comment:
well I believe the reason here could be simplified:
We have to add nullable to the type when we are building the properties, because the "nullibility via optionality" has not been considered.
But when we are in an options bag, this "nullibility via optionality" has been done by the request builder or operation builder, therefore we should do that again here.

Reply:
Updated comments